src/Tribe/Aggregator/Record/Batch_Queue.php

```diff
@@ -0,0 +1,422 @@
+	 */
+	public function __construct( $record, Tribe__Events__Aggregator__Record__Queue_Cleaner $cleaner = null ) {
+		if ( is_numeric( $record ) ) {
+			$record = Tribe__Events__Aggregator__Records::instance()->get_by_post_id( $record );
+		}
+
+		if ( ! is_object( $record ) || ! $record instanceof \Tribe__Events__Aggregator__Record__Abstract ) {
+			$this->null_process = true;
+
+			return;
+		}
+
+		$cleaner = $cleaner ? $cleaner : new Tribe__Events__Aggregator__Record__Queue_Cleaner();
+		$cleaner->remove_duplicate_pending_records_for( $record );
+
+		if ( $cleaner->maybe_fail_stalled_record( $record ) ) {
+			$this->null_process = true;
+
+			return;
+		}
+
+		$this->record = $record;
+		$this->activity();
+	}
+
+	/**
+	 * GEt the activity if a call to a dynamic attribute is taking place in this case `$this->>activity`
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key The dynamic key to be returned.
+	 *
+	 * @return mixed|Tribe__Events__Aggregator__Record__Activity
+	 */
+	public function __get( $key ) {
+		if ( $key === 'activity' ) {
+			return $this->activity();
+		}
+
+		return null;
+	}
+
+	/**
+	 * Returns the activity object for the processing of this Queue.
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed|Tribe__Events__Aggregator__Record__Activity
+	 */
+	public function activity() {
+		if ( empty( $this->activity ) ) {
+			if (
+				empty( $this->record->meta[ self::$activity_key ] )
+				|| ! $this->record->meta[ self::$activity_key ] instanceof Tribe__Events__Aggregator__Record__Activity
+			) {
+				$this->activity = new Tribe__Events__Aggregator__Record__Activity();
+			} else {
+				$this->activity = $this->record->meta[ self::$activity_key ];
+			}
+		}
+
+		return $this->activity;
+	}
+
+	/**
+	 * Allows us to check if the Events Data has still pending
+	 *
+	 * @since TBD
+	 *
+	 * @return boolean
+	 */
+	public function is_fetching() {
+		return $this->is_in_progress();
+	}
+
+	/**
+	 * Shortcut to check how many items are going to be processed next
+	 *
+	 * @since TBD
+	 *
+	 * @return int
+	 */
+	public function count() {
+		return 0;
+	}
+
+	/**
+	 * Shortcut to check if this queue is empty or it has a null process.
+	 *
+	 * @since TBD
+	 *
+	 * @return boolean `true` if this queue instance has acquired the lock and
+	 *                 the count is 0, `false` otherwise.
+	 */
+	public function is_empty() {
+		if ( $this->null_process ) {
+			return true;
+		}
+
+		return ! $this->is_in_progress();
+	}
+
+	/**
+	 * After the process has been completed make sure the `post_modified` and `post_status` are updated accordingly.
+	 *
+	 * @since TBD
+	 *
+	 * @return $this
+	 */
+	protected function complete() {
+		try {
+			// Updates the Modified time for the Record Log.
+			$args = [
+				'ID'            => $this->record->post->ID,
+				'post_modified' => ( new DateTime( 'now', new DateTimeZone( 'UTC' ) ) )->format( Tribe__Date_Utils::DBDATETIMEFORMAT ),
```

Comment:
Use `Tribe__Date_Utils::build_date_object` method so we can avoid the try catch.